datahub-web-react/src/app/entity/shared/siblingUtils.ts

```diff
@@ -51,9 +51,16 @@ const combineMerge = (target, source, options) => {
     return destination;
 };
 
-// use when you want to merge and array of objects by key in the object as opposed to by index of array
+function convertObjectKeysToLowercase(object: Record<string, unknown>) {
+    return Object.fromEntries(Object.entries(object).map(([key, value]) => [key.toLowerCase(), value]));
+}
+
+// use when you want to merge an array of objects by key in the object as opposed to by index of array
 const mergeArrayOfObjectsByKey = (destinationArray: any[], sourceArray: any[], key: string) => {
-    return values(merge(keyBy(destinationArray, key), keyBy(sourceArray, key)));
+    const destination = convertObjectKeysToLowercase(keyBy(destinationArray, key));
```

Comment:
what is the key here? fieldPath?

Reply:
yup! so we have an array of objects, all of these objects have a `fieldPath` entry, and `keyBy` will convert the array of objects into an object where each key is the value of `fieldPath` for each array item, and the value is the original array. That would then convert something like this:
```
[{ fieldPath: 'user_name', ... }, { fieldPath: 'user_id', ... }, { fieldPath: 'user_email', ... }]
```
into this:
```
{
    user_name: { fieldPath: 'user_name', ... },
    user_id: { fieldPath: 'user_id', ... },
    user_email: { fieldPath: 'user_email', ... },
}
```
so that we can pass these two objects into the `merge` function below which will deep merge these objects by key (instead of just index in the array like it was doing before).

Then finally calling the `values` function will return an array of all of the values of the object, resulting in the same shape of the original arrays, but properly merged by key.